### PR TITLE
added getState and getStateMap PyAlbany functions

### PR DESCRIPTION
**Pull request by thartland**
added getState and getStateMap functions in order to be able to access the PDE solution from within PyAlbany application codes.

### CODE REVIEW

**Review by kliegeois — CHANGES_REQUESTED** [review]
Thanks @thartland for the PR.
The PR looks good!

There are just 2 needed modifications and it would be great if you could update one of the PyAlbany tests to get the solution and check it (i.e., compare the mean or compare all the entries of the solution with a solution stored in a file for example).

File: PyAlbany/src/Albany_Interface.cpp [hunk 1]
```diff
@@ -327,6 +345,38 @@ Teuchos::RCP<PyTrilinosVector> PyProblem::getResponse(const int g_index)
...
+    }
+    else
+    {
+#ifdef PYALBANY_DOES_NOT_USE_DEEP_COPY
+        Teuchos::RCP<Thyra_Vector> s = thyraResponses.back();
+        Teuchos::RCP<PyTrilinosVector> s_out = Albany::getTpetraVector(s);
+        return s_out;
+#else
+        std::cout << "PyAlbany Does not use deep copy" << std::endl;
+	      Teuchos::RCP<const Thyra_Vector> s = thyraResponses.back();
```
[the remark this answers is not in the document]
Comment (thartland) [reply PyAlbany/src/Albany_Interface.cpp]: Great, the updates have been made. I will now work on adding the test.

[the remark this answers is not in the document]
Comment (thartland) [reply PyAlbany/src/Albany_Interface.cpp]: @kliegeois I have added a test of the getState and getStateMap functions as suggested. Solution reference data is loaded and compared to that obtained by utilized the aforementioned methods.

**Comment by kliegeois** [comment]
@thartland there is still something strange with the align, I tried to fix it on my computer but the issue remains and it looks like what you did with your initial commit.

The rest looks good, thanks!

**Comment by kliegeois** [comment]
OK got it now, it is fixed.

**Review by kliegeois — APPROVED** [review]

**Pull request merged**